File: app/src/main/java/com/scheffer/erik/popularmovies/moviedatabaseapi/adapters/MoviesAdapter.java
package com.scheffer.erik.popularmovies.moviedatabaseapi.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;

import com.scheffer.erik.popularmovies.R;
import com.scheffer.erik.popularmovies.moviedatabaseapi.dataclasses.Movie;
import com.squareup.picasso.Picasso;

import java.util.List;

import static com.scheffer.erik.popularmovies.moviedatabaseapi.ApiConstants.MOVIES_DATABASE_BASE_POSTER_URL;

public class MoviesAdapter extends ArrayAdapter<Movie> {
    public MoviesAdapter(@NonNull Context context, List<Movie> movies) {
        super(context, 0, movies);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.item_movie, parent, false);
        }

        Movie movie = getItem(position);
        if (movie != null) {
            Picasso.with(getContext())
                    .load(MOVIES_DATABASE_BASE_POSTER_URL + "w780//" + movie.getPosterPath())
                    .into((ImageView) convertView.findViewById(R.id.movie_poster));
        }

        return convertView;
    }
}

File: app/src/main/java/com/scheffer/erik/popularmovies/moviedatabaseapi/adapters/MoviesAdapter.kt
package com.scheffer.erik.popularmovies.moviedatabaseapi.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import com.scheffer.erik.popularmovies.R
import com.scheffer.erik.popularmovies.moviedatabaseapi.MOVIES_DATABASE_BASE_POSTER_URL
import com.scheffer.erik.popularmovies.moviedatabaseapi.models.Movie
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.item_movie.view.*

class MoviesAdapter(context: Context, movies: List<Movie>)
    : ArrayAdapter<Movie>(context, 0, movies) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val returnView = convertView ?: LayoutInflater.from(context).inflate(R.layout.item_movie,
                                                                             parent,
                                                                             false)

        getItem(position)?.let {
            Picasso.with(context)
                    .load(MOVIES_DATABASE_BASE_POSTER_URL + "w780//" + it.posterPath)
                    .into(returnView.movie_poster)
        }

        return returnView
    }
}